Makefile

```diff
@@ -194,6 +194,8 @@ generate-mocks: install-mock-generators
 	mockery --name 'API' --dir="./engine/protocol" --case=underscore --output="./engine/protocol/mock" --outpkg="mock"
 	mockery --name '.*' --dir="./engine/access/state_stream" --case=underscore --output="./engine/access/state_stream/mock" --outpkg="mock"
 	mockery --name 'ConnectionFactory' --dir="./engine/access/rpc/connection" --case=underscore --output="./engine/access/rpc/connection/mock" --outpkg="mock"
+	mockery --name 'Communicator' --structname 'NodeCommunicatorMock' --dir="./engine/access/rpc/backend" --case=underscore --output="./engine/access/rpc/backend" --inpackage
```

Comment:
@peterargue I'm afraid in this case with same params I end up with circular dependencies the mock package will depend on backend and backend from the mock. Therefore the mock has to be inside the backend package. 
Let me know if if this can be solved in some other way :)

Reply:
thanks, will do

Reply:
https://github.com/onflow/flow-go/pull/4598/commits/56e03073d70469707b5b5ec8178a10e21d3834ad